### PR TITLE
CLOUDP-80014: Add Merging Functions for Containers, ContainerPorts and VolumeMounts

### PR DESCRIPTION
**Pull request by chatton**
Based on top of #275

### All Submissions:

* [ ] Have you opened an Issue before filing this PR?
* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: pkg/util/merge/merge.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+import (
+	"sort"
+
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/util/contains"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// StringSlices accepts two slices of strings, and returns a string slice
+// containing the distinct elements present in both. The elements are returned sorted.
+func StringSlices(slice1, slice2 []string) []string {
```
Comment (chatton) [inline pkg/util/merge/merge.go:12]: function moved from unexported utility function into its own package, no changes made

File: pkg/util/merge/merge_test.go [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+import (
+	"reflect"
+	"testing"
+
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/container"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestMergeStringSlices(t *testing.T) {
```
Comment (chatton) [inline pkg/util/merge/merge_test.go:12]: this test was moved with the function

File: pkg/util/merge/merge.go [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+func VolumeMount(original, override corev1.VolumeMount) corev1.VolumeMount {
+	merged := original
+
+	if override.Name != "" {
+		merged.Name = override.Name
+	}
+
+	if override.ReadOnly {
+		merged.ReadOnly = override.ReadOnly
+	}
```
Comment (bznein) [inline pkg/util/merge/merge.go:163]: [q] does this mean we can only override with `true` and never with `false`?

**Review by bznein — APPROVED** [review]
LGTM

File: pkg/util/merge/merge.go (hunk 3 above)
Comment (chatton) [reply pkg/util/merge/merge.go:163]: yes, unfortunately it doesn't look like there is an easy way to allow overrides for falsy values. There is no way to distinguish between a specification of `false` versus the absence of a value.

**Pull request merged**